### PR TITLE
Fixed multi input prompt for MapReduceChain

### LINKED ISSUE #4752: LLM Not Receiving prompt ARGS
### System Info

0.0.170, Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.31, Python 3.9
```python
    Generative_Result_Message = """Given the following schema table, sql query and sql result. Provide a human readable answer to the question
    {sql_answering_document}
    Question: {question}
    Resulting Query: {sql_query}

    
    Return only the answer to the question and create your own human readable answer based off the sql result and sql query
    
    Below is the query result:
    """

    SQL_RESULT_PROMPT = PromptTemplate(
        input_variables=["question", "sql_query", "sql_answering_document"],
        template=Generative_Result_Message,
    )

        generative_result_llm = ChatOpenAI(
            model_name="gpt-4",
            temperature=self.temperature,
            openai_api_key=settings.OPENAI_API_KEY,
            client=get_client(),
        )

        generative_result_llm_chain = LLMChain(
            llm=generative_result_llm, prompt=self.SQL_RESULT_PROMPT
        )

        generative_result_reduce_chain = StuffDocumentsChain(
            llm_chain=generative_result_llm_chain,
            document_variable_name="sql_answering_document",
        )

        combine_documents = MapReduceDocumentsChain(
            llm_chain=generative_result_llm_chain,
            combine_document_chain=generative_result_reduce_chain,
            document_variable_name="sql_answering_document",
        )

        map_reduce = MapReduceChain(
            combine_documents_chain=combine_documents,
            text_splitter=CharacterTextSplitter(),
        )

      result =  map_reduce.run(
      {
                            "question": document["generated_question"],
                            "sql_query": sql_query,
                            "sql_answering_document": "sql_answering_document",
                            "input_text": query_result
                        })
      
```

This is the error log I'm getting 

> 
>     answer = map_reduce(
>   File "/usr/local/lib/python3.9/site-packages/langchain/chains/base.py", line 142, in __call__
>     raise e
>   File "/usr/local/lib/python3.9/site-packages/langchain/chains/base.py", line 136, in __call__
>     self._call(inputs, run_manager=run_manager)
>   File "/usr/local/lib/python3.9/site-packages/langchain/chains/mapreduce.py", line 89, in _call
>     outputs = self.combine_documents_chain.run(
>   File "/usr/local/lib/python3.9/site-packages/langchain/chains/base.py", line 243, in run
>     return self(kwargs, callbacks=callbacks)[self.output_keys[0]]
>   File "/usr/local/lib/python3.9/site-packages/langchain/chains/base.py", line 142, in __call__
>     raise e
>   File "/usr/local/lib/python3.9/site-packages/langchain/chains/base.py", line 136, in __call__
>     self._call(inputs, run_manager=run_manager)
>   File "/usr/local/lib/python3.9/site-packages/langchain/chains/combine_documents/base.py", line 84, in _call
>     output, extra_return_dict = self.combine_docs(
>   File "/usr/local/lib/python3.9/site-packages/langchain/chains/combine_documents/map_reduce.py", line 144, in combine_docs
>     results = self.llm_chain.apply(
>   File "/usr/local/lib/python3.9/site-packages/langchain/chains/llm.py", line 160, in apply
>     raise e
>   File "/usr/local/lib/python3.9/site-packages/langchain/chains/llm.py", line 157, in apply
>     response = self.generate(input_list, run_manager=run_manager)
>   File "/usr/local/lib/python3.9/site-packages/langchain/chains/llm.py", line 80, in generate
>     prompts, stop = self.prep_prompts(input_list, run_manager=run_manager)
>   File "/usr/local/lib/python3.9/site-packages/langchain/chains/llm.py", line 108, in prep_prompts
>     selected_inputs = {k: inputs[k] for k in self.prompt.input_variables}
>   File "/usr/local/lib/python3.9/site-packages/langchain/chains/llm.py", line 108, in <dictcomp>
>     selected_inputs = {k: inputs[k] for k in self.prompt.input_variables}
> KeyError: 'question'
> 

When I inspected the affected library files and log the data before the method that's triggering the error this is the output

> [2023-05-15 23:39:46,775: INFO/ForkPoolWorker-7] kwargs in chains/base.py (part two) kwargs: {'input_documents': [Document(page_content="[('2-3 times a week', 4), ('Twice a month', 2), ('Once a week', 2), ('On occasions', 1)]", metadata={})]} args: ()
> [2023-05-15 23:39:46,775: INFO/ForkPoolWorker-7] inputs from library file: {'input_documents': [Document(page_content="[('2-3 times a week', 4), ('Twice a month', 2), ('Once a week', 2), ('On occasions', 1)]", metadata={})]}
> [2023-05-15 23:39:46,776: INFO/ForkPoolWorker-7] input listings from prep_prompts [{'sql_answering_document': "[('2-3 times a week', 4), ('Twice a month', 2), ('Once a week', 2), ('On occasions', 1)]"}]

Which is weird, I'm not sure why the prompt inputs would not include the one from the llm I setup

### Who can help?

@hwchase17 

### Information

- [ ] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [X] LLMs/Chat Models
- [ ] Embedding Models
- [X] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [X] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

N/A

### Expected behavior

N/A

### PR DESCRIPTION
**Pull request by imeckr**
# Fixed multi input prompt for MapReduceChain

Added `kwargs` support for inner chains of `MapReduceChain` via `from_params` method
Currently the `from_method` method of intialising `MapReduceChain` chain doesn't work if prompt has multiple inputs. It happens because it uses `StuffDocumentsChain` and `MapReduceDocumentsChain` underneath, both of them require specifying `document_variable_name` if `prompt` of their `llm_chain` has more than one `input`.

With this PR, I have added support for passing their respective `kwargs` via the `from_params` method.

## Who can review? 
@dev2049 @hwchase17 @agola11

### CODE REVIEW

**Review by eyurtsev — COMMENTED** [review]
Looks good! Wondering if we could update the type to a Mapping for immutability :)

File: langchain/chains/mapreduce.py [hunk 1]
```diff
@@ -38,15 +38,22 @@ def from_params(
         prompt: BasePromptTemplate,
         text_splitter: TextSplitter,
         callbacks: Callbacks = None,
+        combine_chain_kwargs: Optional[Dict[str, Any]] = None,
```
[the remark this answers is not in the document]
Comment (dev2049) [reply langchain/chains/mapreduce.py:41]: https://docs.python.org/3/library/typing.html#typing.Dict

**Comment by imeckr** [comment]
@dev2049 @hwchase17 @agola11 Can anyone of you review this ?

**Comment by dev2049** [comment]
could we add either an example to the map reduce notebook or an integration test where this is used

**Comment by imeckr** [comment]
@dev2049 Let me know if this example works.

**Review by hwchase17 — APPROVED** [review]

**Pull request merged**